pynicotine/gtkgui/search.py

```diff
@@ -1397,21 +1311,22 @@ def on_download_folders(self, *_args, download_location=""):
             requested_folders[user][folder] = download_location
 
             visible_files = []
-            for row in self.all_data:
+            for result in core.search.searches[self.token].results[user]:
+                file_path = result.file_path
 
                 # Find the wanted directory
-                if folder != row[12].rsplit("\\", 1)[0]:
+                if folder != file_path.rsplit("\\", 1)[0]:
                     continue
 
                 # remove_destination is False because we need the destination for the full folder
                 # contents response later
                 destination = core.transfers.get_folder_destination(user, folder, remove_destination=False)
 
-                (_counter, user, _flag, _h_speed, _h_queue, _directory, _file_type, _filename,
-                    _h_size, h_bitrate, h_length, _bitrate, fullpath, _country_code, size, _speed,
-                    _queue, _length, _has_free_slots) = row
-                visible_files.append(
-                    (user, fullpath, destination, size.get_value(), h_bitrate, h_length))
+                size = result.size
+                h_bitrate, _bitrate, h_length, _length = slskmessages.FileListMessage.parse_result_bitrate_length(
+                    size, result.attributes)
```

Comment:
This doesn't seem to belong here.

Consider processing `result.attributes` and `destination` in `core.search.request_folder_download()` instead...

`visible_files.append((user, file_path, result.size, result.attributes))`